### PR TITLE
Move `force_garbage_collection` from `Study.__init__` to `Study.optimize`.

### PR DESCRIPTION
**Pull request by toshihikoyanase**
This is a follow-up PR for #533. Previously, I create #627, but I find that it can be simplified because `Study.force_garbage_collection` property is only used in `Study.optimize` method.

In addition, it has two additional changes as follows:

- It may be controversial but the argument is renamed to `disable_gc`. `force_garbage_collection` is more accurate but I think users may be worried about using the argument.
- The explanation of `disable_gc` is added according to [this advice](https://github.com/pfnet/optuna/pull/627#discussion_r337849488).

### CODE REVIEW

File: optuna/study.py [hunk 1]
```diff
@@ -214,7 +212,8 @@ def optimize(
             timeout=None,  # type: Optional[float]
             n_jobs=1,  # type: int
             catch=(Exception, ),  # type: Union[Tuple[()], Tuple[Type[Exception]]]
-            callbacks=None  # type: Optional[List[Callable[[Study, structs.FrozenTrial], None]]]
+            callbacks=None,  # type: Optional[List[Callable[[Study, structs.FrozenTrial], None]]]
+            disable_gc=False  # type: bool
```
Comment (g-votte) [inline optuna/study.py]: Though it's out of the original scope of this PR, `disable_gc=False` seems like double negative. How about changing the name like `gc_after_trial`?

**Comment by hvy** [comment]
Just dropping a random thoughts I had after an offline discussion with @g-votte and @toshihikoyanase. If we do want to simplify the interface in the future, maybe we could monitor the memory consumption (e.g. using [`psutil`](https://github.com/giampaolo/psutil with some heuristic interval) to raise warnings when it's increasing telling the user to run the gc manually or perhaps actually run the gc in Optuna.

File: optuna/study.py (hunk 1 above)
Comment (toshihikoyanase) [reply optuna/study.py]: > `gc_after_trial`

Sounds good. I think it is more appropriate than `disable_gc`.
Thank you for your suggestion.

Fixed: https://github.com/pfnet/optuna/pull/632/commits/f384af5a2ad84daff2a360b36d676a6b6803329e.

**Comment by toshihikoyanase** [comment]
In addition to the argument name, I also update the description of `gc_after_trial` in this commit (https://github.com/pfnet/optuna/pull/632/commits/a794edd72b4b589ce81c1fc0ee6bad13a1d14f3f). If you have any comments, please let me know.

**Pull request merged**